src/crypto/index.js

```diff
@@ -275,21 +275,38 @@ Crypto.prototype._checkAndStartKeyBackup = async function() {
             // well that's told us. we won't try again.
             this._checkedForBackup = true;
         }
-        return;
+        return null;
     }
     this._checkedForBackup = true;
 
     const trustInfo = await this.isKeyBackupTrusted(backupInfo);
 
     if (trustInfo.usable && !this.backupInfo) {
-        console.log("Found usable key backup: enabling key backups");
+        console.log(
+            "Found usable key backup v" + backupInfo.version +
+            ": enabling key backups",
+        );
         this._baseApis.enableKeyBackup(backupInfo);
     } else if (!trustInfo.usable && this.backupInfo) {
         console.log("No usable key backup: disabling key backup");
         this._baseApis.disableKeyBackup();
     } else if (!trustInfo.usable && !this.backupInfo) {
         console.log("No usable key backup: not enabling key backup");
+    } else if (trustInfo.usable && this.backupInfo) {
+        // may not be the same version: if not, we should switch
+        if (backupInfo.version !== this.backupInfo.version) {
+            console.log(
+                "On backup version " + this.backupInfo.version + " but found " +
+                "version " + backupInfo.version + ": switching.",
+            );
+            this._baseApis.disableKeyBackup();
+            this._baseApis.enableKeyBackup(backupInfo);
```

Comment:
Hmm, I thought there was a similar path that would switch like this as well, but can't seem to find it now...

Reply:
I don't remember writing it, although possibly we could get away with just the `enableKeyBackup`